kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/inband/DecryptionDekCache.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.encryption.inband;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executor;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.benmanes.caffeine.cache.AsyncLoadingCache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.RemovalCause;
+
+import io.kroxylicious.filter.encryption.FilterThreadExecutor;
+import io.kroxylicious.filter.encryption.dek.CipherSpec;
+import io.kroxylicious.filter.encryption.dek.Dek;
+import io.kroxylicious.filter.encryption.dek.DekManager;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * A cache of DEKs used on the decryption path.
+ * @param <K> The type of KEK id.
+ * @param <E> The type of encrypted DEK.
+ */
+public class DecryptionDekCache<K, E> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DecryptionDekCache.class);
+
+    public static final int NO_MAX_CACHE_SIZE = -1;
+    private final DekManager<K, E> dekManager;
+
+    record CacheKey<E>(
+                       @Nullable CipherSpec cipherSpec,
+                       @Nullable E edek) {
+        public CacheKey {
+            if (cipherSpec == null ^ edek == null) {
+                throw new IllegalArgumentException();
+            }
+        }
+
+        @SuppressWarnings({ "rawtypes", "unchecked" })
+        private static final CacheKey NONE = new CacheKey(null, null);
+
+        @SuppressWarnings("unchecked")
+        static <E> CacheKey<E> none() {
+            return NONE;
+        }
+
+        public boolean isNone() {
```

Comment:
I appreciate this is a refactor (and hence this is a non blocking thought) but I don't really understand `None` as a term. Particularly `cacheKey.isNone()` feels like an odd phrase to read. Having said that `cacheKeys.add(DecryptionDekCache.CacheKey.none());` does read nicely. 

So I guess what I'm asking is should it be `CacheKey.UNENCRYPTED` ? Or `CacheKey.PLAIN_TEXT`?